### ISSUE TITLE
Question about Reproduction: Performance Gap & Sudden Drop in R@1 During Validation

### ISSUE THREAD

**Issue by Frank-Micro** [Delong-liu-bupt/Word4Per#9]
Dear Authors,
First of all, thank you for open-sourcing your excellent work! I am currently reproducing the experiments of FAFA on the SynCPR/ITCPR datasets as reported in your paper, but I encountered some issues and hope to seek your guidance.
1. Gap Between Reproduced Results and Paper Metrics
I followed the configuration and procedures in the repository strictly for training. The key metrics achieved after training are as follows:
{
    "model": "tuned_recall_at1_step.pt",
    "dataset": "itcpr",
    "fda_k": 6,
    "fda_alpha": 0.5,
    "fd_margin": 0.5,
    "recall_at_1": 40.32697677612305,
    "recall_at_5": 61.48955154418945,
    "recall_at_10": 69.75476837158203,
    "mAP": 50.23530197143555
}
There is a noticeable gap between these results and the metrics reported in your paper. I wonder if I missed any details in configuration or data processing.
For reference, I verified that the official released checkpoint reproduces the paper-level performance on my ITCPR setup, which suggests the evaluation pipeline is correct.
2. Abnormal Phenomenon During Training
From the validation_metrics_by_step data, I observed that after training reaches 23,000 steps, the recall@1 drops sharply from a normal level to around 0.045 and fails to recover afterward. This is the core abnormal issue during my reproduction.
3. My Core Training Configuration (Key Parameters)
To facilitate your problem localization, I list the core configurations below:
{
    "backbone": "pretrain",
    "batch_size": 256,
    "blip_model_name": "blip2_fafa_cpr",
    "exp_name": "FAFA_SynCPR_FDA_FD_MFR",
    "fd_margin": 0.5,
    "fda_alpha": 0.5,
    "fda_k": 6,
    "itcpr_root": "/home/u2025170870/FAFA/data/ITCPR",
    "json_path": "/home/u2025170870/FAFA/data/SynCPR/SynCPR.json",
    "learning_rate": 2e-06,
    "loss_fd": 1.0,
    "loss_fda": 1.0,
    "loss_mfr": 0.5,
    "num_epochs": 10,
    "num_workers": 4,
    "resume_path": null,
    "save_best": true,
    "save_last": false,
    "save_memory": false,
    "save_training": true,
    "setting": [
        "annotations"
    ],
    "syncpr_data_path": "/home/u2025170870/FAFA/data/SynCPR",
    "target_ratio": 1.25,
    "transform": "squarepad",
    "validation_frequency": 1,
    "validation_step": 500
}
Questions for You:
Given the above metrics not matching the paper’s results, did I miss any key configurations (e.g., pretrained weight version, data augmentation details, optimizer parameters)?
Have you encountered the sudden drop in R@1 after 23,000 steps? What might be the possible causes (e.g., learning rate scheduling, overfitting, data loading issues)?
Are there any crucial details for reproducing this work that are not mentioned in the README but essential?
Thank you very much for taking the time to answer my questions. Looking forward to your reply!
Best regards,

**Comment by Delong-liu-bupt** [comment]
Hi,

I carefully checked the raw experimental records from the paper version, and the configuration is basically consistent with what is reported in the README. However, there are two details that differ:

1. I used a smaller `validation_step` of 200.
2. I found that in the training for the paper version, I only used a **partial** subset of the generated data (I have just updated `paper_used.json` and the experimental performance logs from that time to the HuggingFace repository).

The full dataset generation wasn't completed at that time. I later attempted to train on the full data and achieved an R1 of around 45%, so the optimal R1 performance shouldn't degrade as severely as you observed. Regarding your specific low performance, I don't have a definitive answer, but one thing is certain: with such massive data, there must be data points inconsistent with the test set distribution. Data filtering could likely achieve better results.

Regarding the other issue, most of the time the optimal performance occurs **before** 23,000 steps; I rarely ran it that far. However, reviewing my logs, I found that I encountered similar sudden drops in early experiments when using larger learning rates (like 1e-4 to 1e-5). After lowering the learning rate to 2e-6, I didn't see that happen again.

I suggest you try the updated `paper_used.json` (aligned with the paper version) to see if the performance loss persists, or try your own data filtering. An important reason I updated the full data was to allow everyone to freely attempt different combinations.

Hope this helps.

**Comment by Frank-Micro** [comment]
Hi!
Thank you so much for your detailed and helpful reply—this clears up a lot of confusion for me!
I’ll first try using the updated paper_used.json with the original configuration, and adjust the validation_step to 200 as you mentioned. I’ll also double-check the learning rate setting (I did use 2e-6, but I’ll confirm it’s correctly applied in the training script).
I really appreciate you sharing the experimental logs and insights—it’s a huge help for my reproduction work.
I’ll update you with the results once I finish the new training runs!
Best regards,
